nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/remote/StandardRemoteProcessGroup.java

```diff
@@ -856,10 +864,75 @@ public void refreshFlowContents() throws CommunicationsException {
         }
     }
 
+    @Override
+    public String getNetworkInterface() {
+        readLock.lock();
+        try {
+            return networkInterfaceName;
+        } finally {
+            readLock.unlock();
+        }
+    }
+
+    @Override
+    public void setNetworkInterface(final String interfaceName) {
+        writeLock.lock();
+        try {
+            this.networkInterfaceName = interfaceName;
+
+            try {
+                final Enumeration<InetAddress> inetAddresses = NetworkInterface.getByName(interfaceName).getInetAddresses();
+
+                if (inetAddresses.hasMoreElements()) {
+                    this.localAddress = inetAddresses.nextElement();
```

Comment:
I guess my thought is that the user is selecting a name that results in a mapping. What if one of the objects within the enumeration is at some point invalid, doesn't that break the user's assumption that he or she would always get a valid InetAddress that is bound to that network interface name?